File: 1743/1743.c
int restoreTheArrayFromAdjacentPai(struct TreeNode* root) {
    if (!root) return 0;
    int l = restoreTheArrayFromAdjacentPai(root->left);
    int r = restoreTheArrayFromAdjacentPai(root->right);
    return 1 + (l > r ? l : r);
}

File: 1743/1743.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var restoreTheArrayFromAdjacentPai = function(root) {
    if (!root) return 0;
    return 1 + Math.max(restoreTheArrayFromAdjacentPai(root.left), restoreTheArrayFromAdjacentPai(root.right));
};